openpathsampling/pathmover.py

```diff
@@ -546,225 +538,105 @@ def __call__(self, trial):
 
         return trials
 
-    def _shoot(self, shooting_point, ensemble):
-        """Implementation of the shooting
-
-        Parameters
-        ----------
-        shooting_point : ShootingPoint
-            the initial shooting point instance containing a reference to the
-            initial trajectory
-        ensemble : Ensemble
-            the ensemble for which the trajectory is to be created. This defines
-            the stopping criterion
-
-        Returns
-        -------
-        ShootingPoint
-            the final shooting point referencing the final trajectory
-        """
-        return shooting_point
-
-
-class ForwardShootMover(ShootMover):
-    """A forward shooting sample generator
-    """
-    def _shoot(self, shooting_point, ensemble):
-        shoot_str = "Shooting {sh_dir} from frame {fnum} in [0:{maxt}]"
-        logger.info(shoot_str.format(fnum=shooting_point.index,
-                                     maxt=len(shooting_point.trajectory)-1,
-                                     sh_dir="forward",
-                                    ))
-
-        # Run until one of the stoppers is triggered
-        partial_trajectory = self.engine.generate(
-            shooting_point.snapshot.copy(),
-            running = [
-                paths.PrefixTrajectoryEnsemble(
-                    ensemble,
-                    shooting_point.trajectory[0:shooting_point.index]
-                ).can_append
-            ]
-        )
-
-        trial_trajectory = \
-            shooting_point.trajectory[0:shooting_point.index] + \
-            partial_trajectory
-
-        trial_point = paths.ShootingPoint(
-            shooting_point.selector,
-            trial_trajectory,
-            shooting_point.index
-        )
+    def _make_forward_trajectory(self, trajectory, shooting_index):
+        initial_snapshot = trajectory[shooting_index].copy()
+        run_f = paths.PrefixTrajectoryEnsemble(self.target_ensemble, 
+                                               trajectory[0:shooting_index]
+                                              ).can_append
+        partial_trajectory = self.engine.generate(initial_snapshot, 
+                                                  running=[run_f])
+        trial_trajectory = trajectory[0:shooting_index] + partial_trajectory
+        return trial_trajectory
 
-        return trial_point
+    def _make_backward_trajectory(self, trajectory, shooting_index):
+        initial_snapshot = trajectory[shooting_index].reversed_copy()
```

Comment:
Same thing as before. And if we can use a shallow copy, then the `.reversed` property is the best way to get the reversed snapshot and not create a redundant copy.